src/Files.App/Actions/Favorites/PinItemAction.cs

```diff
@@ -3,53 +3,82 @@
 using Files.App.Commands;
 using Files.App.Contexts;
 using Files.App.Extensions;
+using Files.App.Filesystem;
 using Files.App.ServicesImplementation;
+using Files.App.UserControls.Widgets;
+using System.ComponentModel;
 using System.Linq;
 using System.Threading.Tasks;
+using Windows.Storage;
 
 namespace Files.App.Actions.Favorites
 {
 	internal class PinItemAction : ObservableObject, IAction
 	{
-		public IContentPageContext context = Ioc.Default.GetRequiredService<IContentPageContext>();
+		private readonly IContentPageContext context = Ioc.Default.GetRequiredService<IContentPageContext>();
+		private readonly IQuickAccessService service = Ioc.Default.GetRequiredService<IQuickAccessService>();
 
-		private readonly IQuickAccessService quickAccessService = Ioc.Default.GetRequiredService<IQuickAccessService>();
+		public string Label { get; } = "PinToFavorites".GetLocalizedResource();
 
-		public string Label { get; } = "BaseLayoutItemContextFlyoutPinToFavorites/Text".GetLocalizedResource();
+		public RichGlyph Glyph { get; } = new(opacityStyle: "ColorIconPinToFavorites");
 
-		public RichGlyph Glyph { get; } = new RichGlyph(opacityStyle: "ColorIconPinToFavorites");
+		private bool isExecutable;
+		public bool IsExecutable => isExecutable;
 
-		public bool IsExecutable
+		public PinItemAction()
 		{
-			get
-			{
-				if ((context.SelectedItems.Any() && context.SelectedItems.All(x => !x.IsPinned))
-					|| (context.Folder is not null && !context.Folder.IsPinned))
-					return true;
+			isExecutable = GetIsExecutable();
 
-				return false;
-			}
+			context.PropertyChanged += Context_PropertyChanged;
+			App.QuickAccessManager.UpdateQuickAccessWidget += QuickAccessManager_DataChanged;
 		}
 
-		public PinItemAction()
+		public async Task ExecuteAsync()
 		{
-			context.PropertyChanged += Context_PropertyChanged;
+			if (context.HasSelection)
+			{
+				var items = context.SelectedItems.Select(x => x.ItemPath).ToArray();
+				await service.PinToSidebar(items);
+			}
+			else if (context.Folder is not null)
+			{
+				await service.PinToSidebar(context.Folder.ItemPath);
```

Comment:
I think we should remove the curly braces

Reply:
It's done.

Reply:
We usually use curly braces if there is an else statement.

Reply:
Oh, I wasn't aware of that

Reply:
I put the style back as before.